direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BlobLogReader.java

```diff
@@ -152,16 +152,23 @@ protected BlobLogReader(BlobStorageAccessor accessor, Context context) {
   @SuppressWarnings("unchecked")
   @Override
   public List<Partition> getPartitions(long startStamp, long endStamp) {
-    List<Partition> ret = new ArrayList<>();
-    AtomicInteger id = new AtomicInteger();
-    @Nullable BulkStoragePartition<BlobT> current = null;
-    Stream<Path> paths = fs.list(startStamp, endStamp).sorted();
-    for (Iterator<Path> it = paths.iterator(); it.hasNext(); ) {
-      current =
-          considerBlobForPartitionInclusion(((BlobPathT) it.next()).getBlob(), id, current, ret);
+    final List<Partition> ret = new ArrayList<>();
+    final AtomicInteger id = new AtomicInteger();
+    @SuppressWarnings("rawtypes")
+    Stream<BlobPath<BlobT>> paths = (Stream) fs.list(startStamp, endStamp);
+    paths =
+        paths.sorted(
+            Comparator.comparingLong(
+                path -> namingConvention.parseMinMaxTimestamp(path.getBlobName()).getFirst()));
```

Comment:
Can we stabilize the sort even more? I'd suggest sorting by name when min timestamps equal.

Reply:
I think we should take a `fileSize` into account, than we can probably a smarter merging 🤔

Reply:
If we want to keep the partitions sorted by timestamps I doubt we can do anything better than sort it by filename.